src/internal/packager/components.go

```diff
@@ -0,0 +1,207 @@
+package packager
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/defenseunicorns/zarf/src/config"
+	"github.com/defenseunicorns/zarf/src/internal/message"
+	"github.com/defenseunicorns/zarf/src/internal/utils"
+	"github.com/defenseunicorns/zarf/src/types"
+	"github.com/pterm/pterm"
+	"gopkg.in/yaml.v2"
+)
+
+const horizontalRule = "───────────────────────────────────────────────────────────────────────────────────────"
+
+func getValidComponents(allComponents []types.ZarfComponent, requestedComponentNames []string) []types.ZarfComponent {
+	var validComponentsList []types.ZarfComponent
+	var orderedKeys []string
+	var choiceComponents []string
+
+	componentGroups := make(map[string][]types.ZarfComponent)
+
+	// Break up components into choice groups
+	for _, component := range allComponents {
+		key := component.Choice
+		// If not a choice group, then use the component name as the key
+		if key == "" {
+			key = component.Name
+		} else {
+			// Otherwise, add the component name to the choice group list for later validation
+			choiceComponents = appendIfNotExists(choiceComponents, component.Name)
+		}
+
+		// Preserve component order
+		orderedKeys = appendIfNotExists(orderedKeys, key)
+
+		// Append the component to the list of components in the group
+		componentGroups[key] = append(componentGroups[key], component)
+	}
+
+	// Loop through each component group in original order and handle required, requested or user confirmation
+	for _, key := range orderedKeys {
+
+		componentGroup := componentGroups[key]
+
+		// Choice groups are handled differently for user confirmation
+		userChoicePrompt := len(componentGroup) > 1
+
+		// Loop through the components in the group
+		for _, component := range componentGroup {
+			// First check if the component is required or requested via CLI flag
+			requested := isRequiredOrRequested(component, requestedComponentNames)
+
+			// If the user has not requested this component via CLI flag, then prompt them if not a choice group
+			if !requested && !userChoicePrompt {
+				requested = confirmOptionalComponent(component)
+			}
+
+			if requested {
+				// Mark deployment as appliance mode if this is an init config and the k3s component is enabled
+				if component.Name == "k3s" && config.IsZarfInitConfig() {
```

Comment:
we have a `k8s.DistroIsK3s` const we can use here to check against instead of using a hardcoded string